csrc/python_frontend/fusion_record.h

```diff
@@ -1823,7 +1823,7 @@ struct SelectOpRecord : RecordFunctor {
 
   void operator()(FusionState& fd) final {
     auto arg1 = fd.getFusionState(args_.at(0).index)->template as<TensorView>();
-    auto arg3 = fd.getFusionState(args_.at(1).index)->template as<TensorView>();
+    auto arg3 = fd.getFusionState(args_.at(1).index);
```

Comment:
nitpick: can we add a test that triggers this?

Reply:
This fails with the Debug mode:

```
pytest -s -v tests/python/test_python_frontend.py -k test_select 
```

```
    self._finalize_definition()
RuntimeError:  INTERNAL ASSERT FAILED at "/raid/nmaruyama/debug1/csrc/python_frontend/fusion_state.cpp":124, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues.
Detected exception while building Fusion Ir. The failing RecordFunctor is: T2 = fd.ops.select(T0, S1, dim=0)
NvFuser error message is:  INTERNAL ASSERT FAILED at "/raid/nmaruyama/debug1/csrc/utils.h":119, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues.
Exception raised from as at /raid/nmaruyama/debug1/csrc/utils.h:119 (most recent call first):
frame #0: nvfuser::nvfCheckFail(char const*, char const*, unsigned int, char const*) + 0x92 (0x7f07c2458642 in /raid/nmaruyama/debug1/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #1: <unknown function> + 0xa82e4b (0x7f07c201be4b in /raid/nmaruyama/debug1/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #2: <unknown function> + 0xa82be4 (0x7f07c201bbe4 in /raid/nmaruyama/debug1/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #3: <unknown function> + 0xaf75d3 (0x7f07c20905d3 in /raid/nmaruyama/debug1/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #4: <unknown function> + 0x15fc11e (0x7f07c2b9511e in /raid/nmaruyama/debug1/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #5: nvfuser::python_frontend::FusionDefinition::finalizeDefinition() + 0x32d (0x7f07c2b8de3d in /raid/nmaruyama/debug1/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #6: <unknown function> + 0x8e7aa9 (0x7f07c1e80aa9 in /raid/nmaruyama/debug1/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #7: <unknown function> + 0x8e7a89 (0x7f07c1e80a89 in /raid/nmaruyama/debug1/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #8: <unknown function> + 0x8e7a41 (0x7f07c1e80a41 in /raid/nmaruyama/debug1/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #9: <unknown function> + 0x8e79ba (0x7f07c1e809ba in /raid/nmaruyama/debug1/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #10: <unknown function> + 0x8e7889 (0x7f07c1e80889 in /raid/nmaruyama/debug1/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #11: <unknown function> + 0x9e8195 (0x7f07c1f81195 in /raid/nmaruyama/debug1/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #12: /opt/conda/pytorch/bin/python() [0x4fcaf7]
```

Reply:
@naoyam ran the test in `DEBUG` mode to find the error. The CI runs in `RELEASE` mode. The existing test already triggers the error.

Reply:
Ha, thanks for that extra info. I'm good with it as-is then.

Reply:
Just confirmed the test passes with this change.